test_open_notebook fails when running tests twice in a row

Running the tests locally seems fine when you run them once, but when running them a second time, `test_open_notebook` fails with a `FileExistsError`:
```
    @flaky(max_runs=3)
    def test_open_notebook(qtbot):
        """Test that a notebook can be opened from a non-ascii directory."""
        # Move the test file to non-ascii directory
        test_notebook = osp.join(LOCATION, 'test.ipynb')
        test_notebook_non_ascii = osp.join(LOCATION, u'äöüß', 'test.ipynb')
>       os.mkdir(os.path.join(LOCATION, u'äöüß'))
E       FileExistsError: [Errno 17] File exists: '/home/jitse/misc/spyder-notebook/spyder_notebook/tests/äöüß'

spyder_notebook/tests/test_plugin.py:163: FileExistsError
```
The first test run creates this file, which interferes with the second test run. This does not affect the CI runs because the CI servers delete the environment after every run.

--- a/spyder_notebook/tests/test_plugin.py
+++ b/spyder_notebook/tests/test_plugin.py
@@ -11,6 +11,7 @@ import json
 import os
 import os.path as osp
 import shutil
+import tempfile
 
 # Third-party library imports
 from flaky import flaky
@@ -19,10 +20,10 @@ import requests
 from qtpy.QtWebEngineWidgets import WEBENGINE
 from qtpy.QtCore import Qt, QTimer
 from qtpy.QtWidgets import QFileDialog, QApplication, QLineEdit
+from spyder.config.base import get_home_dir
 
 # Local imports
 from spyder_notebook.notebookplugin import NotebookPlugin
-
 
 #==============================================================================
 # Constants
@@ -156,12 +157,15 @@ def test_shutdown_notebook_kernel(qtbot):
 
 
 @flaky(max_runs=3)
-def test_open_notebook(qtbot, tmpdir):
+def test_open_notebook(qtbot):
     """Test that a notebook can be opened from a non-ascii directory."""
     # Move the test file to non-ascii directory
     test_notebook = osp.join(LOCATION, 'test.ipynb')
-    test_notebook_non_ascii = osp.join(str(tmpdir), u'äöüß', 'test.ipynb')
-    os.mkdir(os.path.join(str(tmpdir), u'äöüß'))
+
+    # For Python 2, non-ascii directory needs to be under home dir
+    tmpdir = tempfile.mkdtemp(dir=get_home_dir())
+    test_notebook_non_ascii = osp.join(tmpdir, u'äöüß', 'test.ipynb')
+    os.mkdir(os.path.join(tmpdir, u'äöüß'))
     shutil.copyfile(test_notebook, test_notebook_non_ascii)
 
     # Create notebook

--- a/spyder_notebook/tests/test_plugin.py
+++ b/spyder_notebook/tests/test_plugin.py
@@ -25,17 +25,17 @@ from spyder.config.base import get_home_dir
 # Local imports
 from spyder_notebook.notebookplugin import NotebookPlugin
 
-#==============================================================================
+# =============================================================================
 # Constants
-#==============================================================================
+# =============================================================================
 NOTEBOOK_UP = 5000
 INTERACTION_CLICK = 100
 LOCATION = osp.realpath(osp.join(os.getcwd(), osp.dirname(__file__)))
 
 
-#==============================================================================
+# =============================================================================
 # Utility functions
-#==============================================================================
+# =============================================================================
 def prompt_present(nbwidget):
     """Check if an In prompt is present in the notebook."""
     if WEBENGINE:
@@ -98,9 +98,9 @@ def is_kernel_up(kernel_id, sessions_url):
     return kernel
 
 
-#==============================================================================
+# =============================================================================
 # Fixtures
-#==============================================================================
+# =============================================================================
 @pytest.fixture
 def setup_notebook(qtbot):
     """Set up the Notebook plugin."""
@@ -111,9 +111,9 @@ def setup_notebook(qtbot):
     return notebook
 
 
-#==============================================================================
+# =============================================================================
 # Tests
-#==============================================================================
+# =============================================================================
 @flaky(max_runs=3)
 def test_hide_header(qtbot):
     """Test that the kernel header is hidden."""
@@ -125,17 +125,17 @@ def test_hide_header(qtbot):
     qtbot.waitUntil(lambda: prompt_present(nbwidget), timeout=NOTEBOOK_UP)
 
     # Wait for hide header
-    qtbot.waitUntil(lambda: text_present(nbwidget,
-                                         'id="header-container" class="hidden"'),
+    html_fragment = 'id="header-container" class="hidden"'
+    qtbot.waitUntil(lambda: text_present(nbwidget, html_fragment),
                     timeout=NOTEBOOK_UP)
 
     # Assert that the header is hidden
-    assert text_present(nbwidget, 'id="header-container" class="hidden"')
+    assert text_present(nbwidget, html_fragment)
 
 
 @flaky(max_runs=3)
 def test_shutdown_notebook_kernel(qtbot):
-    """Test that the kernel is shutdown from the server when closing a notebook."""
+    """Test that kernel is shutdown from server when closing a notebook."""
     # Create notebook
     notebook = setup_notebook(qtbot)
 
@@ -145,7 +145,8 @@ def test_shutdown_notebook_kernel(qtbot):
 
     # Get kernel id for the client
     client = notebook.get_current_client()
-    qtbot.waitUntil(lambda: client.get_kernel_id() is not None, timeout=NOTEBOOK_UP)
+    qtbot.waitUntil(lambda: client.get_kernel_id() is not None,
+                    timeout=NOTEBOOK_UP)
     kernel_id = client.get_kernel_id()
     sessions_url = client.get_session_url()
 
@@ -216,7 +217,8 @@ def test_save_notebook(qtbot, tmpdir):
 
     # Assert that the In prompt has "test" in it
     # and the client has the correct name
-    qtbot.waitUntil(lambda: text_present(nbwidget, text="test"), timeout=NOTEBOOK_UP)
+    qtbot.waitUntil(lambda: text_present(nbwidget, text="test"),
+                    timeout=NOTEBOOK_UP)
     assert text_present(nbwidget, text="test")
     assert notebook.get_current_client().get_short_name() == "save"
 

--- a/spyder_notebook/tests/test_plugin.py
+++ b/spyder_notebook/tests/test_plugin.py
@@ -7,9 +7,10 @@
 """Tests for the plugin."""
 
 # Standard library imports
+import json
 import os
 import os.path as osp
-import json
+import shutil
 
 # Third-party library imports
 from flaky import flaky
@@ -155,13 +156,13 @@ def test_shutdown_notebook_kernel(qtbot):
 
 
 @flaky(max_runs=3)
-def test_open_notebook(qtbot):
+def test_open_notebook(qtbot, tmpdir):
     """Test that a notebook can be opened from a non-ascii directory."""
     # Move the test file to non-ascii directory
     test_notebook = osp.join(LOCATION, 'test.ipynb')
-    test_notebook_non_ascii = osp.join(LOCATION, u'äöüß', 'test.ipynb')
-    os.mkdir(os.path.join(LOCATION, u'äöüß'))
-    os.rename(test_notebook, test_notebook_non_ascii)
+    test_notebook_non_ascii = osp.join(str(tmpdir), u'äöüß', 'test.ipynb')
+    os.mkdir(os.path.join(str(tmpdir), u'äöüß'))
+    shutil.copyfile(test_notebook, test_notebook_non_ascii)
 
     # Create notebook
     notebook = setup_notebook(qtbot)
@@ -179,7 +180,7 @@ def test_open_notebook(qtbot):
 
 
 @flaky(max_runs=3)
-def test_save_notebook(qtbot):
+def test_save_notebook(qtbot, tmpdir):
     """Test that a notebook can be saved."""
     # Create notebook
     notebook = setup_notebook(qtbot)
@@ -201,7 +202,7 @@ def test_save_notebook(qtbot):
     qtbot.keyClick(nbwidget, Qt.Key_QuoteDbl, delay=INTERACTION_CLICK)
 
     # Save the notebook
-    name = 'save.ipynb'
+    name = osp.join(str(tmpdir), 'save.ipynb')
     QTimer.singleShot(1000, lambda: manage_save_dialog(qtbot, fname=name))
     notebook.save_as(name=name)
 

